### PR TITLE
GCE: Bump addon resizer to 1.8.9 version and use metrics to get cluster size there

### PR DESCRIPTION
**Pull request by wojtek-t**
```release-note
NONE
```

### CODE REVIEW

File: cluster/addons/metrics-server/metrics-server-deployment.yaml [hunk 1]
```diff
@@ -64,7 +64,7 @@ spec:
           name: https
           protocol: TCP
       - name: metrics-server-nanny
-        image: k8s.gcr.io/addon-resizer:1.8.7
+        image: staging-k8s.gcr.io/addon-resizer:1.8.9
```
Comment (x13n) [inline cluster/addons/metrics-server/metrics-server-deployment.yaml:67]: Why use staging container image? This should be promoted to regular k8s.gcr.io and only then used.

Comment (wojtek-t) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml:67]: It's why this is WIP PR :-)

**Comment by wojtek-t** [comment]
@bskiba @mm4tt @x13n - it's now ready for review

**Comment by wojtek-t** [comment]
/hold

We want to enable measuring resource usage for addon-resize in perf-tests repo before we merge that.

File: cluster/addons/metrics-server/metrics-server-deployment.yaml [hunk 2]
```diff
@@ -64,7 +64,7 @@ spec:
           name: https
           protocol: TCP
       - name: metrics-server-nanny
-        image: k8s.gcr.io/addon-resizer:1.8.7
+        image: us.gcr.io/k8s-artifacts-prod/autoscaling/addon-resizer:1.8.9
```
Comment (x13n) [inline cluster/addons/metrics-server/metrics-server-deployment.yaml:67]: Why isn't this available in global gcr.io?

**Review by x13n — COMMENTED** [review]
/lgtm

File: cluster/addons/metrics-server/metrics-server-deployment.yaml (hunk 2 above)
Comment (mm4tt) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml:67]: +1, we should use k8s.gcr.io, image needs to be promoted there. I'll send you instructions how to do it

Comment (bskiba) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml:67]: I'm interested in how to do it as well, shouldn't it happen automatically?

Comment (wojtek-t) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml:67]: @mm4tt - the instruction you sent me are outdate - they used to be valid before we deprecated the old repository

@x13n - I wasn't the one setting that up, but it seems it's not only the case for autoscaling - we seem to no longer promoting images to global gcr - you might want to follow up with people that were setting that up:
https://github.com/kubernetes/k8s.io/tree/master/k8s.gcr.io

Comment (mm4tt) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml:67]: They are deprecated, but still applicable, as k8s.gcr.io vanity domain flip hasn't happened yet, see https://github.com/kubernetes/release/issues/270

Comment (wojtek-t) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml:67]: So why we we're even allowing for pushing images to the new repo? That means pushing to two different places - seems strange at least...

Comment (wojtek-t) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml]: So no matter how ridiculous it is, I updated to the k8s.gcr.io repo.

**Comment by wojtek-t** [comment]
@mm4tt @x13n - PTAL

**Comment by wojtek-t** [comment]
We have the metrics from the last run: https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/1266052058390204420/artifacts/ResourceUsageSummary_load_2020-05-29T04:52:05Z.json

unholing

/hold cancel

/retest

**Pull request merged**